rollup/circuitscapacitychecker/libzkp/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+#![feature(once_cell)]
+
+pub mod checker {
+    use crate::utils::{bool_to_int, c_char_to_vec};
+    use libc::c_char;
+    use std::cell::OnceCell;
+    use std::panic;
+    use types::eth::BlockTrace;
+    use zkevm::capacity_checker::CircuitCapacityChecker;
+
+    static mut CHECKER: OnceCell<CircuitCapacityChecker> = OnceCell::new();
+
+    /// # Safety
+    #[no_mangle]
+    pub unsafe extern "C" fn new_circuit_capacity_checker() {
+        env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("debug"))
+            .format_timestamp_millis()
+            .init();
+
+        let c = CircuitCapacityChecker::new();
+        CHECKER.set(c).unwrap();
+    }
+
+    /// # Safety
+    #[no_mangle]
+    pub unsafe extern "C" fn reset_circuit_capacity_checker() {
+        CHECKER.get_mut().unwrap().reset()
+    }
+
+    /// # Safety
+    #[no_mangle]
+    pub unsafe extern "C" fn apply_tx(tx_traces: *const c_char) -> c_char {
+        let tx_traces_vec = c_char_to_vec(tx_traces);
+        let traces = serde_json::from_slice::<BlockTrace>(&tx_traces_vec).unwrap();
+        let result = panic::catch_unwind(|| {
+            CHECKER
+                .get_mut()
+                .unwrap()
+                .estimate_circuit_capacity(&[traces])
+                .unwrap()
+        });
+        if result.is_err() {
+            return 0 as c_char; // other errors than circuits capacity overflow
+        }
+        let (acc_row_usage, tx_row_usage) = result.unwrap();
+        if acc_row_usage.is_ok {
+            return 1 as c_char; // row usage ok
```

Comment:
Using `0` for no error seems more conventional

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/321/commits/fe0defd08a1bbbc23a97fc0a75d4ee5090ce1340